### DISCUSSION TITLE
Blockscad like interface and component store

### DISCUSSION THREAD

**Discussion by WolfgangFahl** [jscad/OpenJSCAD.org#819] [Ideas]
Every once in a while I still do blockscad designs instead of OpenJscad ones due to the simplicity.

Recently I did a [handle for my vice](http://wiki.bitplan.com/index.php/Vice_Handle). I used an M12x280  screw and added a plastic knob at the end.

![grafik](https://user-images.githubusercontent.com/1336221/116775531-a7e1a800-aa63-11eb-8a4a-9058547168e6.png)
![grafik](https://user-images.githubusercontent.com/1336221/116775534-ad3ef280-aa63-11eb-97e7-097b75ffc501.png)

The blockscad design looked like this:
![grafik](https://user-images.githubusercontent.com/1336221/116775492-8b457000-aa63-11eb-82f2-1cd5639b050d.png)

For an extended version of this it would be great to add a thread to the knob so there could be another knob at the other end of the handle instead of fixing it there in position as i do.

I do have a scad design for threads. 

Now the issue is that three parts:
- scad design
- blockscad
- openjscad

do not have proper intereoperability between each other and there is no "component/module" store to use.

I envision to be able to use blockscad with external module definitions using openjscad picking modules from a module store where the module store will allow to use traditional scad designs.

In the end I'd love to show off my own design in my wiki and share them this way.

If we'd overcome the problems involved the world would IMHO be a better place.

**Comment by hrgdavor** [comment]
Modules for jscad should be published to npm, and that way it will be possible to use them in jscad and blockscad etc.

Regarding a screw module I am currently in progress of creating one.
http://3d.hrg.hr/jscad/three/threejscad.html?uri=#model.screw.js

That is a page i use to develop it, ao it is not read yet to be used in jscad. 
I want to make some more improvements and also add support for both outside and inside screw. 

I want to make inside grooves directly without booleans (booleans are great but inherently much slower)

**Comment by z3dev** [comment]
Here's a Web UI that has a blocks interface. And this one is maintained as well.

https://makecode.buildbee.com/
